Jack/Property.fs

```diff
@@ -43,6 +43,13 @@ module Property =
         else
             Failure [] |> ofResult
 
+    let private convert (value : 'T) =
+        match box value with
+        | :? Lazy<bool> as b -> ofBool b.Value
+        | :? Property   as p -> p
+        | :? bool       as b -> ofBool b
+        | _                  -> ofBool true
```

Comment:
Sorry, but I don't like these dynamic checks, if we implement an implication operator then it has to be using the FsControl style ad-hoc polymorphism. The fact that everything still type checks even if I pass an `int` to `convert` is a problem for me.

I'm not sure that this is necessarily required though, I have something in mind for implication which uses computation expressions and should work for both your examples, I'll put up a PR and see what you think?